src/components/closed-captions/closed-captions.tsx

```diff
@@ -43,8 +48,50 @@ const ClosedCaptions = connect(mapStateToProps)(
           setCCOn(activeTextTrack?.language !== 'off');
         }, [activeTextTrack]);
 
+        useEffect(() => {
+          registerToBottomBar(COMPONENT_NAME, player, () => registerComponent());
+        }, []);
+
+      const registerComponent = (): any => {
+        return {
+          ariaLabel: () => getAriaLabel(),
+          displayName: COMPONENT_NAME,
+          order: 5,
+          svgIcon: () => getSvgIcon(),
+          onClick: () => onClick(),
+          component: () => {
+            return getComp({...props, classNames: [style.upperBarIcon]});
+          },
+          selfManagement: true
```

Comment:
the name "selfManagement" is not very clear (actually "selfManaged" is already clearer). what is the value used for ?

Reply:
renamed to `shouldHandleOnClick`